app/user-setup/index.tsx

```diff
@@ -0,0 +1,215 @@
+import { setUpUser } from "@/firebase/functions";
+import { SetUpUserRequestType, SetUpUserRequestSchema } from "@/types/user";
+import { useForm, Controller } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { UserContext } from "@/contexts/userContext";
+import { CampusEnum } from "@/enums/campus";
+import useUploadImage from "@/hooks/useUploadImage";
+import { useRouter } from "expo-router";
+import { clsx } from "clsx";
+
+import {
+    View,
+    Text,
+    TouchableOpacity,
+    ActivityIndicator,
+    TextInput,
+    Image,
+} from "react-native";
+import { Divider } from "@ui-kitten/components";
+import { AVATAR_IMAGES_FOLDER } from "@/constants/storage";
+
+function UserSetupPage() {
+    const { user } = useContext(UserContext);
+    const [isLoading, setIsLoading] = useState(false);
+    const { uploadImages } = useUploadImage();
+    const router = useRouter();
+
+    const [campusSelected, setCampusSelected] = useState(
+        CampusEnum.Values["Miagao Campus"].toString(),
+    );
+    const [imageURI, setImageURI] = useState(user?.photoURL ?? "");
+    const {
+        handleSubmit,
+        formState: { errors },
+        control,
+        setValue,
+    } = useForm<SetUpUserRequestType>({
+        resolver: zodResolver(SetUpUserRequestSchema),
+        defaultValues: {
+            name: user?.displayName ?? "",
+            imageUrl: imageURI,
+            campus: CampusEnum.Values["Miagao Campus"],
+        },
+    });
+    async function openImageLibrary() {
+        try {
+            let res = await ImagePicker.launchImageLibraryAsync({
+                mediaTypes: ["images"],
+                allowsMultipleSelection: false,
+                aspect: [9, 16],
+                quality: 1,
+            });
+
+            if (!res.canceled) {
+                setValue("imageUrl", res.assets[0].uri);
+                setImageURI(res.assets[0].uri);
+            }
+        } catch (error) {
+            console.error(error);
+        }
+    }
+
+    async function openCamera() {
+        try {
+            let res = await ImagePicker.launchCameraAsync({
+                mediaTypes: ["images"],
+                aspect: [9, 16],
+                quality: 1,
+            });
+
+            if (!res.canceled) {
+                setValue("imageUrl", res.assets[0].uri);
+                setImageURI(res.assets[0].uri);
+            }
+        } catch (error) {
+            console.error(error);
+        }
+    }
+
+    function handleSetCampus(campus: string) {
+        setValue(
+            "campus",
+            CampusEnum.options[0] === campus
+                ? CampusEnum.options[0]
+                : CampusEnum.options[1],
+        );
+        setCampusSelected(campus);
+    }
+
+    async function handleSetupUser(data: SetUpUserRequestType) {
+        setIsLoading(true);
+        try {
+            await uploadImages([imageURI], AVATAR_IMAGES_FOLDER);
+            await setUpUser(data);
```

Comment:
we should modifly data to have the proper url because as of now this is what's being recorded in the db:
![image](https://github.com/user-attachments/assets/b99f4cbb-90b8-476c-98f9-a5f5ee2c6a88)

it should be
```tsx
data.imageUrl = await uploadImages([imageURI], AVATAR_IMAGES_FOLDER)[0];
await setUpUser(data);
```

Reply:
ahh oo bitaw, na miss nako gege